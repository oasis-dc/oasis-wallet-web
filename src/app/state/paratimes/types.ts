import { StringifiedBigInt } from 'types/StringifiedBigInt'
import { ErrorPayload } from 'types/errors'
import { ParaTime } from '../../../config'

export enum TransactionFormSteps {
  TransferType = 'transferType',
  ParaTimeSelection = 'paraTimeSelection',
  TransactionRecipient = 'transactionRecipient',
  TransactionAmount = 'transactionAmount',
  TransactionConfirmation = 'transactionConfirmation',
  TransactionSummary = 'transactionSummary',
  TransactionError = 'transactionError',
}

export enum TransactionTypes {
  Deposit = 'deposit',
  Withdraw = 'withdraw',
}

export interface TransactionForm {
  amount: string
  confirmTransfer: boolean
  confirmTransferToValidator: boolean
  confirmTransferToForeignAccount: boolean
  defaultFeeAmount: string
  // compatible with oasisRT.signatureSecp256k1
  ethPrivateKey: string
  // provided by user and used in form inputs allowing back and forth form navigation
  ethPrivateKeyRaw: string
  feeAmount: string
  feeGas: string
  paraTime?: ParaTime
  recipient: string
  type: TransactionTypes | undefined
}

export interface ParaTimesState {
  balance: StringifiedBigInt
  isLoading: boolean
  transactionError?: ErrorPayload
  transactionForm: TransactionForm
  transactionFormStep: TransactionFormSteps
}

export type Runtime = {
  address: string
  id: string
  decimals: number
  feeGas: bigint
}

export type ParaTimeTransaction = Pick<
  TransactionForm,
  'amount' | 'ethPrivateKey' | 'feeAmount' | 'feeGas' | 'recipient' | 'type'
>
