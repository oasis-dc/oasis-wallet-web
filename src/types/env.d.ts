type Backend = 'oasismonitor' | 'oasisscanV2' | 'nexus'

declare global {
  namespace NodeJS {
    export interface ProcessEnv {
      REACT_APP_BACKEND: Backend
      REACT_APP_TRANSAK_URL: string
      REACT_APP_TRANSAK_PARTNER_ID: string
      REACT_APP_LOCALNET: '1' | undefined
      REACT_APP_E2E_TEST: '1' | undefined
      NODE_ENV: 'development' | 'production' | 'test'
    }
  }

  interface Window {
    REACT_APP_BACKEND: Backend
  }
}

export {}
