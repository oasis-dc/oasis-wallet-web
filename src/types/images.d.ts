declare module '*.png' {
  /** Image URL */
  const url: string
  export default url
}

declare module '*.svg' {
  /** Image URL */
  const url: string
  export default url
}

declare module 'bundle-text:*.svg' {
  /** Image content */
  const content: string
  export default content
}
