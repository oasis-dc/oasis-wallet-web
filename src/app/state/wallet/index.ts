import { PayloadAction, current } from '@reduxjs/toolkit'
import { createSlice } from 'utils/@reduxjs/toolkit'

import {
  BalanceUpdatePayload,
  OpenFromPrivateKeyPayload,
  OpenSelectedAccountsPayload,
  Wallet,
  WalletState,
} from './types'

export const initialState: WalletState = {
  wallets: {},
  selectedWallet: undefined,
}

export const walletSlice = createSlice({
  name: 'wallet',
  initialState,
  reducers: {
    openWalletFromMnemonic(state, action: PayloadAction<OpenSelectedAccountsPayload>) {},
    openWalletFromPrivateKey(state, action: PayloadAction<OpenFromPrivateKeyPayload>) {},
    openWalletsFromLedger(state, action: PayloadAction<OpenSelectedAccountsPayload>) {},
    deleteWallet(state, action: PayloadAction<string>) {
      delete state.wallets[action.payload]
    },
    selectFirstWallet(state) {
      state.selectedWallet = Object.values(current(state.wallets))[0].address
    },
    selectWallet(state, action: PayloadAction<string | undefined>) {
      state.selectedWallet = action.payload
    },
    fetchWallet(state, action: PayloadAction<Wallet>) {},
    updateBalance(state, action: PayloadAction<BalanceUpdatePayload>) {
      // We simply ignore the updateBalance action in rare cases when opening a
      // second tab while account is loading:
      // - tab2 initializes
      // - tab1 dispatches updateBalance
      // - tab2 asks to sync state
      // - tab2 receives updateBalance for an account it doesn't have
      // - tab1 responds with state
      // - tab2 receives synced state with the account it didn't have
      //
      // Alternatives:
      // - Remove updateBalance from whitelistTabSyncActions and have an
      //   outdated balance in case tab2 makes a transaction.
      // - Refactor redux-state-sync to ignore actions until tab2 receives
      //   synced state. But tab1 needs to listen to actions too, even tho it
      //   never asks for synced state.
      if (state.wallets[action.payload.address]) {
        state.wallets[action.payload.address].balance = {
          ...state.wallets[action.payload.address].balance,
          ...action.payload.balance,
        }
      }
    },
    walletOpened(state, action: PayloadAction<Wallet>) {
      const newWallet = action.payload
      state.wallets[newWallet.address] = newWallet
      state.selectedWallet ??= newWallet.address
    },
    setWalletName(state, action: PayloadAction<{ address: string; name: string }>) {
      state.wallets[action.payload.address].name = action.payload.name
    },
  },
})

export const { actions: walletActions } = walletSlice
