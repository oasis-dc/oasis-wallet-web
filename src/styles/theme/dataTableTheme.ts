import { IDataTableStyles, ITheme } from 'react-data-table-component'

export const dataTableTheme = {
  background: { default: 'unset' },
  highlightOnHover: {
    default: '#88888833',
    text: 'unset',
  },
  text: {
    primary: 'unset',
    secondary: 'unset',
    disabled: 'unset',
  },
  sortFocus: {
    default: 'unset',
  },
  divider: { default: '#AAAAAAaa' },
} as Partial<ITheme>

export const dataTableCustomStyles: IDataTableStyles = {
  headCells: {
    style: {
      fontSize: '16px',
    },
  },
  rows: {
    style: {
      fontSize: '16px',
    },
  },
}
