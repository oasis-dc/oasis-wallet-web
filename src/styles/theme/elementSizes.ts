export const sidebarSmallSizeLogo = 32
export const sidebarMediumSizeLogo = 40
export const sidebarLargeSizeLogo = 40
export const mobileToolbarZIndex = 3
export const mobileToolbarButtonsZIndex = mobileToolbarZIndex + 1
export const buildBannerZIndex = mobileToolbarZIndex - 1

export const mobileFooterNavigationHeight = '4rem'
