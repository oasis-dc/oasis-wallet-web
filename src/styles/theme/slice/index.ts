import { PayloadAction } from '@reduxjs/toolkit'
import { createSlice } from 'utils/@reduxjs/toolkit'

import { getThemeFromStorage, saveTheme } from '../utils'
import { ThemeState } from './types'

export const getInitialState = (): ThemeState => ({
  selected: getThemeFromStorage() || 'system',
})

export const themeSlice = createSlice({
  name: 'theme',
  initialState: () => getInitialState(),
  reducers: {
    changeTheme(state, action: PayloadAction<'dark' | 'light' | 'system'>) {
      saveTheme(action.payload)
      state.selected = action.payload
    },
  },
})

export const { actions: themeActions } = themeSlice
