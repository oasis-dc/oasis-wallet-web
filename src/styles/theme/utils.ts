import { ThemeKeyType } from './slice/types'

/* istanbul ignore next line */
export const isSystemDark = window?.matchMedia
  ? window.matchMedia('(prefers-color-scheme: dark)')?.matches
  : undefined

export const getTargetTheme = (theme: 'light' | 'dark' | 'system') =>
  theme === 'system' ? (isSystemDark ? 'dark' : 'light') : theme

export function saveTheme(theme: ThemeKeyType) {
  window.localStorage.setItem('selectedTheme', theme)
}

/* istanbul ignore next line */
export function getThemeFromStorage(): ThemeKeyType | null {
  return window.localStorage.getItem('selectedTheme') as ThemeKeyType
}
