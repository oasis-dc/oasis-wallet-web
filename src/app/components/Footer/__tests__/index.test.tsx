import { render, screen } from '@testing-library/react'
import { ResponsiveContext } from 'grommet/es6/contexts/ResponsiveContext'
import { MemoryRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import { ThemeProvider } from 'styles/theme/ThemeProvider'
import { configureAppStore } from 'store/configureStore'
import { Wallet } from 'app/state/wallet/types'

import { Footer } from '..'

jest.unmock('react-i18next')

const renderComponent = (store: any, responsiveContext: string) =>
  render(
    <Provider store={store}>
      <MemoryRouter>
        <ThemeProvider>
          <ResponsiveContext.Provider value={responsiveContext}>
            <Footer />
          </ResponsiveContext.Provider>
        </ThemeProvider>
      </MemoryRouter>
    </Provider>,
  )

describe('<Footer />', () => {
  let store: ReturnType<typeof configureAppStore>

  beforeEach(() => {
    jest.resetModules()

    store = configureAppStore({
      wallet: {
        selectedWallet: 'dummy',
        wallets: {
          dummy: {
            address: 'dummy',
          } as Wallet,
        },
      },
    })
  })

  it('should render a link with commit sha after toggling', async () => {
    renderComponent(store, 'large')

    const commitLink = await screen.findByRole('link', { name: /sha0000/i })
    expect(commitLink).toHaveAttribute(
      'href',
      'https://github.com/oasisprotocol/wallet/commit/sha0000000000000000000000000000000000000',
    )
  })

  it('should render a link with version number', () => {
    renderComponent(store, 'large')

    expect(screen.getByRole('link', { name: '1.0.0-dev.1' })).toHaveAttribute(
      'href',
      'https://github.com/oasisprotocol/wallet/releases/tag/v1.0.0-dev.1',
    )
  })

  it('should render mobile version of footer', () => {
    renderComponent(store, 'small')

    expect(screen.getByRole('contentinfo')).toMatchSnapshot()
  })
})
