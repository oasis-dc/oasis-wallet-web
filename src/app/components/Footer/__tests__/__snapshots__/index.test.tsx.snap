// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render mobile version of footer 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-top: auto;
  margin-bottom: 0px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding-left: 24px;
  padding-right: 24px;
  padding-top: 1rem;
  border-radius: 5px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-left: 6px;
  margin-right: 6px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  height: 0;
}

.c3 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 14px;
  line-height: 20px;
  color: #0500e2;
  font-weight: 600;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

.c3:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c2 {
  margin-bottom: 12px;
  font-size: 14px;
  line-height: 20px;
  text-align: center;
}

.c5 {
  margin-top: 12px;
  font-size: 14px;
  line-height: 20px;
  text-align: center;
}

@media only screen and (max-width:768px) {
  .c0 {
    margin-top: auto;
  }
}

@media only screen and (max-width:768px) {
  .c0 {
    margin-bottom: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c0 {
    padding-left: 12px;
    padding-right: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c0 {
    padding-top: 1rem;
  }
}

@media only screen and (max-width:768px) {
  .c6 {
    margin-left: 3px;
  }
}

@media only screen and (max-width:768px) {
  .c6 {
    margin-right: 3px;
  }
}

<footer
  class="c0"
  data-testid="footer"
>
  <div
    class="c1"
    style="flex: 1; align-self: center;"
  >
    <span
      class="c2"
      style="display: inline-flex; flex-direction: column;"
    >
      <span>
        <span>
          ROSE Wallet is 
        </span>
        <a
          class="c3"
          href="https://github.com/oasisprotocol/wallet/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <span>
            open source
          </span>
        </a>
        <span>
           and powered by
        </span>
        <br />
        <span>
          Nexus API & Oasis gRPC
        </span>
      </span>
    </span>
  </div>
  <div
    class="c4"
  />
  <div
    class="c1"
    style="flex: 1; align-self: center;"
  >
    <span
      class="c5"
    >
      <div
        class="c1"
        style="display: inline-flex; flex-direction: row;"
      >
        <a
          class="c3"
          href="https://wallet.oasis.io/t-c.pdf"
          rel="noopener noreferrer"
          target="_blank"
        >
          <span>
            Terms and Conditions
          </span>
        </a>
        <div
          class="c6"
          style="display: inline-block;"
        >
          |
        </div>
      </div>
      <span>
        <span>
          Version: 
        </span>
        <a
          class="c3"
          href="https://github.com/oasisprotocol/wallet/releases/tag/v1.0.0-dev.1"
          rel="noopener noreferrer"
          target="_blank"
        >
          1.0.0-dev.1
        </a>
        <br />
        <span>
           (commit: 
        </span>
        <a
          class="c3"
          href="https://github.com/oasisprotocol/wallet/commit/sha0000000000000000000000000000000000000"
          rel="noopener noreferrer"
          target="_blank"
        >
          sha0000
        </a>
        <span>
          ) built at 1/1/1970, 12:00:00 AM
        </span>
      </span>
    </span>
  </div>
  <div
    class="c4"
  />
  <span
    class="c5"
  >
    <span>
      Weâ€™d love your feedback at 
    </span>
    <a
      class="c3"
      href="mailto:wallet@oasisprotocol.org"
    >
      <span>
        wallet@oasisprotocol.org
      </span>
    </a>
  </span>
</footer>
`;
